cli/Sources/TuistKit/Services/Cache/CacheStartCommandService.swift

```diff
@@ -0,0 +1,80 @@
+@preconcurrency import FileSystem
+import Foundation
+import GRPCCore
+import GRPCNIOTransportHTTP2
+import Path
+import TuistCAS
+import TuistCore
+import TuistLoader
+import TuistRootDirectoryLocator
+import TuistServer
+import TuistSupport
+
+enum CacheStartCommandServiceError: Equatable, LocalizedError {
+    case missingFullHandle
+    
+    var errorDescription: String? {
+        switch self {
+        case .missingFullHandle:
+            return "The 'Tuist.swift' file is missing a fullHandle. See how to set up a Tuist project at: https://docs.tuist.dev/en/server/introduction/accounts-and-projects#projects"
+        }
+    }
+}
+
+struct CacheStartCommandService {
+    private let configLoader: ConfigLoading
+    private let serverEnvironmentService: ServerEnvironmentServicing
+    private let fileSystem: FileSysteming
+
+    init(
+        configLoader: ConfigLoading = ConfigLoader(),
+        serverEnvironmentService: ServerEnvironmentServicing = ServerEnvironmentService(),
+        fileSystem: FileSysteming = FileSystem()
+    ) {
+        self.configLoader = configLoader
+        self.serverEnvironmentService = serverEnvironmentService
+        self.fileSystem = fileSystem
+    }
+
+    func run(
+        path: AbsolutePath
+    ) async throws {
+        let config = try await configLoader.loadConfig(path: path)
+
+        guard let fullHandle = config.fullHandle else {
+            throw CacheStartCommandServiceError.missingFullHandle
+        }
+        let socketPath = Environment.current.stateDirectory
```

Comment:
Initially, I wanted this to be in the `Tuist/Derived` directory relative to the path of the project. Unfortunately, the path of the socket can be only up to 100 characters. While it can be relative here, we would still get an error from xcodebuild when the path set as `COMPILATION_CACHE_REMOTE_SERVICE_PATH` was longer than 100 characters.

So, the safest option is to pick a directory that's close to the root, I ended up choosing the state directory.